pandas/core/indexes/interval.py

```diff
@@ -293,24 +293,22 @@ def __contains__(self, key):
 
     def contains(self, key):
         """
-        Return a boolean indicating if the key is IN the index
-
-        We accept / allow keys to be not *just* actual
-        objects.
+        Return a boolean mask whether the key is contained in the Intervals
+        of the index.
 
         Parameters
         ----------
-        key : int, float, Interval
+        key : scalar, Interval
 
         Returns
         -------
-        boolean
+        boolean array
         """
-        try:
-            self.get_loc(key)
-            return True
-        except KeyError:
-            return False
+        if isinstance(key, Interval):
+            raise TypeError('contains not defined for two intervals')
```

Comment:
I think this should be supported eventually, so this could maybe be a `NotImplementedError`, but don't have strong feelings here.